functions/hosts_with_pe_profile.pp

```diff
@@ -0,0 +1,40 @@
+# Function: hosts_with_pe_profile
+#
+# Queries PuppetDB for hosts with the specified Puppet Enterprise profile.
+# Used by this module to query Puppet Enterprise API endpoints.
+
+# Parameters:
+#
+# $profile: the short name of the Puppet Enterprise profile.
+
+# Results:
+#
+# $hosts: an array of certnames.
+#
+# Replaces the certname with '127.0.0.1' when the certname matches the localhost.
+# Returns ['127.0.0.1'] when PuppetDB returns no hosts.
+
+function puppet_metrics_collector::hosts_with_pe_profile($profile) {
+  if $settings::storeconfigs {
+    $_profile = capitalize($profile)
+    $hosts = puppetdb_query("resources[certname] {
```

Comment:
I wonder if this should have specified `environment` to only return hosts in the current environment?